portal/views/external_assets.py

```diff
@@ -2,31 +2,79 @@
 
 Primarily LifeRay content access - via exposed APIs for portal side caching
 """
-from flask import Blueprint, current_app
-import requests
+from flask import Blueprint, current_app, request
+from requests.exceptions import ConnectionError
+from ..models.app_text import localize_url, time_request
 
 asset_api = Blueprint('asset', __name__)
 
 
+def get_request(url, params=None):
+    """helper for fetching response data from requested URL
+
+    :param url: the URL to pull content from
+    :param params: optional, if provided, use as
+        parameters to the requested URL
+
+    """
+    error_msg = ''
+    try:
+        response = time_request(url, params).json()
+    except ValueError:  # raised when no json is available in response
+        if response.status_code == 200:
+            current_app.logger.warning(
+                "Request did not return json: {}".format(url))
+            return response.text
+        else:
+            error_msg = (
+                "Could not retrieve remote content - " "{} {}".format(
+                    response.status_code, response.reason))
+    except ConnectionError:
+        error_msg = (
+                    "Could not retrieve remove content - Server could not be "
+                    "reached")
+    if error_msg:
+        current_app.logger.error(error_msg + ": {}".format(url))
+    return response, error_msg
+
+
 @asset_api.route('/api/asset/tag/<tag>', methods=('GET',))
 def by_tag(tag):
     url = "{}/c/portal/truenth/asset/query".format(
         current_app.config["LR_ORIGIN"])
+    url = localize_url(url, request.args.get('locale_code', 'en'))
     params = {
         'allTags': tag,
-        'returnFirst': True,
+        'index': request.args.get('index', 0),
+        'content': 'true',
         'version': 'latest',
     }
-    return requests.get(
-        url, params=params).json()['asset']
+    response, error_msg = get_request(url, params)
+    if error_msg:
+        return error_msg
+    if isinstance(response, dict):
+        if 'results' not in response or not len(response['results']):
+            abort(404, 'Remote content not found for tag {}'.format(tag))
```

Comment:
I don't think we should change the Liferay response code if our eventual goal is to have the frontend retrieve content directly from Liferay via AJAX.

If we want `404`s we should ask Victor to change the Liferay API ("find first content item by tag") to `404` on no content item found

Reply:
Agreed.